tests/test_views.py

```diff
@@ -0,0 +1,258 @@
+from urllib.parse import urlencode
+
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Car
+
+
+class PublicViewsTests(TestCase):
+    def test_index_view_login_required(self) -> None:
+        url = reverse("taxi:index")
+        res = self.client.get(url)
+
+        self.assertNotEqual(res.status_code, 200)
```

Comment:
You should check if status code is equal to forbidden instead. Now the test will pass even is server error occurred. Same for the next tests

Reply:
Probably we should check for 302 status code, cause if user unauthorized, django will redirect user to login page with 302 status, also checking for 401/403 doesn't work

Reply:
I agree!